src/Core/Content/Product/Helper/ProductVariationHelper.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Product\Helper;
+
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Content\Property\Aggregate\PropertyGroupOption\PropertyGroupOptionEntity;
+
+class ProductVariationHelper
+{
+    public function buildVariation(ProductEntity $product): array
```

Comment:
```suggestion
    public function build(ProductEntity $product): array
```

It would be better if you directly add the generated variations to the product and return the modified product.

Reply:
I think it might be better not to return the product, since it will be passed by reference anyway.